### PR TITLE
user resource migrated

### PR DESCRIPTION
**Pull request by Sakshi-dell**
User resource migrated with all test cases passing.

image

### CODE REVIEW

File: redfish/provider/resource_redfish_user_account.go [hunk 1]
```diff
@@ -1,5 +1,412 @@
...
+// Schema defines the schema for the resource.
+func (r *UserAccountResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "This Terraform resource is used to manage user entity of the iDRAC Server. We can create, read, modify and delete an existing user using this resource.",
+		Description:         "This Terraform resource is used to manage user entity of the iDRAC Server. We can create, read, modify and delete an existing user using this resource.",
+		Version:             1,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "The ID of the user. Cannot be updated.",
+				Description:         "The ID of the user. Cannot be updated.",
```
Comment (rounak-adhikary) [inline redfish/provider/resource_redfish_user_account.go:59]: The ID of this resource.

File: redfish/provider/resource_redfish_user_account.go [hunk 2]
```diff
@@ -1,5 +1,412 @@
...
+					stringvalidator.LengthBetween(4, 40),
+				},
+			},
+			"redfish_server": schema.SingleNestedAttribute{
+				MarkdownDescription: "Redfish Server",
+				Description:         "Redfish Server",
+				Required:            true,
+				Attributes:          RedfishServerSchema(),
+			},
+			"role_id": schema.StringAttribute{
```
Comment (rounak-adhikary) [inline redfish/provider/resource_redfish_user_account.go:91]: Probably better to add "None" in the Default field, rt?

File: redfish/provider/resource_redfish_user_account.go (hunk 1 above)
Comment (Sakshi-dell) [reply redfish/provider/resource_redfish_user_account.go]: Updated.

File: redfish/provider/resource_redfish_user_account.go (hunk 2 above)
Comment (Sakshi-dell) [reply redfish/provider/resource_redfish_user_account.go:93]: Updated

**Review by Krishnan-Priyanshu — DISMISSED** [review]

File: redfish/provider/provider.go [hunk 3]
```diff
@@ -99,6 +99,8 @@ func (*redfishProvider) Resources(_ context.Context) []func() resource.Resource
 	return []func() resource.Resource{
 		NewPowerResource,
 		NewUserAccountResource,
+		NewSimpleUpdateResource,
+		NewDellIdracAttributesResource,
```
Comment (rounak-adhikary) [inline redfish/provider/provider.go:103]: Is this entry correct?

Comment (rounak-adhikary) [reply redfish/provider/provider.go:103]: I mean, "NewDellIdracAttributesResource"?

Comment (Krishnan-Priyanshu) [reply redfish/provider/provider.go:103]: yes, Idrac attributes resource migrated to tpf in previous pr.

**Review by rounak-adhikary — APPROVED** [review]
LGTM

**Review by Krishnan-Priyanshu — APPROVED** [review]

**Pull request merged**